contracts/farm-bot.sol

```diff
@@ -46,22 +49,29 @@ contract FarmBot is Owned, FarmbotERC20 {
     constructor(
         address _owner,
         address _stakingRewards,
+	address _stakingToken,
         address _revoBounty,
         address _router,
-        address[] memory _path0,
-        address[] memory _path1,
+	address[] memory _rewardsTokens,
+	address[][2][] memory _paths,
         string memory _symbol
     ) Owned(_owner) {
-        stakingRewards = StakingRewards(_stakingRewards);
-        rewardsToken = stakingRewards.rewardsToken();
+        stakingRewards = IMoolaStakingRewards(_stakingRewards);
+
+	for (uint i=0; i<_rewardsTokens.length; i++) {
+	    rewardsTokens.push(IERC20(_rewardsTokens[i]));
+	}
+
+	assert(_paths.length == _rewardsTokens.length);
+	paths = _paths;
```

Comment:
This copying of a dynamic nested array is weird... Solidity has (_had_? no idea...) a bunch of weird constraints around nested arrays. This does compile, but I'm not sure if we'll hit runtime errors or not.